docs/gateway/integrations/langchain.mdx

```diff
@@ -0,0 +1,505 @@
+---
+title: "LangChain Integration"
+sidebarTitle: "LangChain"
+description: "Integrate Helicone AI Gateway with LangChain to access 100+ LLM providers with unified observability."
+"twitter:title": "LangChain Integration - Helicone OSS LLM Observability"
+iconType: "solid"
+---
+
+## Introduction
+
+[LangChain](https://www.langchain.com/) is a popular open-source framework for building applications with large language models across Python, TypeScript, and other languages. By integrating Helicone AI Gateway with LangChain, you can:
+
+- **Route to different models & providers** with automatic failover through a single endpoint
+- **Unified billing** with pass-through billing or bring your own keys
+- **Monitor all requests** with automatic cost tracking in one dashboard
+- **Stream responses** with full observability for real-time applications
+
+<Note>
+  This integration requires only **two changes** to your existing LangChain code - updating the base URL and API key.
+</Note>
+
+## Integration Steps
+
+<Steps>
+  <Step title="Create an account + Generate an API Key">
+    Sign up at [helicone.ai](https://www.helicone.ai) and generate an [API key](https://us.helicone.ai/settings/api-keys).
+
+    <Note>
+      You'll also need to configure your provider API keys (OpenAI, Anthropic, etc.) at [Helicone Providers](https://us.helicone.ai/providers) for BYOK (Bring Your Own Keys).
+    </Note>
+  </Step>
+
+  <Step title="Set environment variables">
+    ```bash
+    # Your Helicone API key
+    export HELICONE_API_KEY=<your-helicone-api-key>
+    ```
+
+    Create a `.env` file in your project:
+    ```env
+    HELICONE_API_KEY=sk-helicone-...
+    ```
+  </Step>
+
+  <Step title="Install LangChain packages">
+    <CodeGroup>
+    ```bash TypeScript
+    npm install @langchain/openai @langchain/core dotenv
+    # or
+    yarn add @langchain/openai @langchain/core dotenv
+    ```
+
+    ```bash Python
+    pip install langchain-openai langchain-core python-dotenv
+    ```
+    </CodeGroup>
+  </Step>
+
+  <Step title="Configure LangChain with Helicone AI Gateway">
+    <CodeGroup>
+    ```typescript TypeScript
+    import { ChatOpenAI } from "@langchain/openai";
+    import { HumanMessage, SystemMessage } from "@langchain/core/messages";
+    import dotenv from 'dotenv';
+
+    dotenv.config();
+
+    // Initialize ChatOpenAI with Helicone AI Gateway
+    const chat = new ChatOpenAI({
+        model: 'gpt-4.1-mini',  // Any model from Helicone registry
+        temperature: 0.8,
```

Comment:
Would remove these additional attributes since they're optional, that way the code looks easier to implement